Client.Core/Flux/Internal/IFluxResultMapper.cs

```diff
@@ -14,5 +15,7 @@ public interface IFluxResultMapper
         /// <typeparam name="T">Type of DomainObject</typeparam>
         /// <returns>Converted DomainObject</returns>
         T ConvertToEntity<T>(FluxRecord fluxRecord);
+
+        object ConvertToEntity(FluxRecord fluxRecord, Type type);
```

Comment:
This is a breaking change. We have to increase major version of the client and add proper description of the change in `CHANGELOG.md`... or avoid this breaking change. 

Please also add documentation comments.

Reply:
this is a tough one... I could create a new Interface called `IFluxResultMapperOfType` or something and then check in the `?`QueryAsync` method if the `IFluxResultMapper` is inheritable from `IFluxResultMapperOfType` and if not throw a new exception which says something like 
> IFluxResultMapper needs also implementing IFluxResultMapperOfType to call this method
but this feels quite hackish, what do you think?

Reply:
> but this feels quite hackish, what do you think?

Yeah, you are right. Let's increate major version in `*.csproj` to version `3.0.0`.